### PR TITLE
Fix double render

### PR DESCRIPTION
**Pull request by azizpunjani**
Since sl-radio `input` is within a label, a `for` attribute is not required according to the spec.

### CODE REVIEW

File: addon/components/sl-select.js [hunk 1]
```diff
@@ -43,6 +43,13 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
     disableSearch: false,
 
     /**
+     * Unique input id that will generated and set on init of component
```
Comment (azizpunjani) [inline addon/components/sl-select.js]: Need to fix typo here.

File: addon/components/sl-textarea.js [hunk 2]
```diff
@@ -110,6 +110,13 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
     spellcheck: Spellcheck.Direction,
 
     /**
+     * Unique textarea id that will generated and set on init of component
```
Comment (azizpunjani) [inline addon/components/sl-textarea.js]: Fix typo

File: addon/templates/components/sl-select.hbs [hunk 3]
```diff
@@ -13,7 +13,7 @@
 {{/if}}
 
 {{input
-    viewName="sl-select-view"
+    id=inputId
```
Comment (notmessenger) [inline addon/templates/components/sl-select.hbs]: Thinking "inputId" should at least be "selectId" to keep with the naming convention used by `sl-textarea`, though what about using something more generic instead?

Going with the generic, 2 out of 3 of these components need this functionality, which is identical, so what about moving it into the `sl-input-based` mixin?

Comment (azizpunjani) [inline addon/templates/components/sl-select.hbs]: Using inputId consistently now for both `sl-textarea` and `sl-select`

**Comment by azizpunjani** [comment]
@notmessenger Moved inputId generation to `sl-input-based`.

**Comment by notmessenger** [comment]
Research use of elementId

**Comment by azizpunjani** [comment]
@notmessenger did some research, this is best approach we have. I initially thought we could use elementId + '-input' in the view itself but handlebars doesn't like that, it's better to put it in a function like we have above.

File: addon/templates/components/sl-select.hbs (hunk 3 above)
Comment (notmessenger) [inline addon/templates/components/sl-select.hbs]: RESOLVED

**Pull request merged**